great_expectations/core/expectation_configuration.py

```diff
@@ -1286,9 +1283,12 @@ def to_json_dict(self) -> dict:
         # NOTE - JPC - 20191031: migrate to expectation-specific schemas that subclass result with properly-typed
         # schemas to get serialization all-the-way down via dump
         myself["kwargs"] = convert_to_json_serializable(myself["kwargs"])
-        myself["expectation_context"] = convert_to_json_serializable(
-            myself["expectation_context"]
-        )
+
+        # Post dump hook removes this value if null so we need to ensure applicability before conversion
+        if "expectation_context" in myself:
+            myself["expectation_context"] = convert_to_json_serializable(
+                myself["expectation_context"]
+            )
```

Comment:
@post_dump cleans this completely so we need to verify the key's existence before conversion.

Could also simply do:
```python
for attr in myself.__dict__:
    myself[attr] = convert_to_json_serializeable(attr)
```
Don't know if that has any implications so kept it simple.